server.js

```diff
@@ -151,6 +152,66 @@ camp.notfound(/.*/, function(query, match, end, request) {
 
 // Vendors.
 
+/**
+ * Registers a new service in the shields.io system. `serviceClass` must be a
+ * class that extends `BaseService`.
+ */
+function registerService(serviceClass) {
```

Comment:
Could this be a method on `BaseService` instead?

Reply:
You mean something like `AppVeyorService.register(camp)`? I think that's doable! I'll take a look on the weekend.

Reply:
Yes, exactly!